### ISSUE TITLE
Optional dynamic segments aren't allowed

### ISSUE THREAD

**Issue by brandonpittman** [kiliman/remix-flat-routes#30]
I'm trying to do optional dynamic segments like this: `/($foo)`

This spits out an error saying the route prefix must come first.

Tried putting the `$` first and got this for the output route:

```
<Route path="reports/:(type)/:id" file="routes/reports_.$(type).$id/_route.tsx" />
```

**Comment by corygibbons** [comment]
I don't know if it's intended but I was able to get it working with `(:type)`, eg. `reports.(:type).$id.tsx`.

**Comment by brandonpittman** [comment]
> I don't know if it's intended but I was able to get it working with `(:type)`, eg. `reports.(:type).$id.tsx`.

Seems like a happy accident. 😆

**Comment by kiliman** [comment]
It's a bug in flat routes... it expects the segment to start with `$`. Optional segments change that, so should support `($foo)`. I'll try and get this fixed today.

**Comment by kiliman** [comment]
Fixed in v0.5.5
